apiserver/pkg/modelservice/modelservice.go

```diff
@@ -0,0 +1,325 @@
+/*
+Copyright 2023 KubeAGI.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package modelservice
+
+import (
+	"context"
+	graphqlutils "github.com/kubeagi/arcadia/apiserver/pkg/utils"
+	"github.com/kubeagi/arcadia/pkg/embeddings"
+	"github.com/kubeagi/arcadia/pkg/utils"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/kubeagi/arcadia/api/base/v1alpha1"
+	"github.com/kubeagi/arcadia/apiserver/graph/generated"
+	"github.com/kubeagi/arcadia/apiserver/pkg/common"
+	"github.com/kubeagi/arcadia/pkg/llms"
+)
+
+// FIXME: Functions duplicated with llm & embedder creation.
+func obj2Embedder(ctx context.Context, c dynamic.Interface, obj *unstructured.Unstructured) *generated.Embedder {
```

Comment:
If this is duplicated code,we can move `ObjectToEmbedder` into `apiserver/pkg/common/object_convertor.go`

Reply:
I think it would be a good idea to expose the `object2model` function for each resource under apiserver/pkg.